File: db4o.net/drs/Db4objects.Drs.Tests/native/Db4objects.Drs.Tests/Foundation/ObjectSetCollection4FacadeTestCase.cs
/* This file is part of the db4o object database http://www.db4o.com

Copyright (C) 2004 - 2009  Versant Corporation http://www.versant.com

db4o is free software; you can redistribute it and/or modify it under
the terms of version 3 of the GNU General Public License as published
by the Free Software Foundation.

db4o is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License along
with this program.  If not, see http://www.gnu.org/licenses/. */
namespace Db4objects.Drs.Tests.Foundation
{
	public class ObjectSetCollection4FacadeTestCase : Db4oUnit.ITestCase
	{
		public static void Main(string[] args)
		{
			new Db4oUnit.ConsoleTestRunner(typeof(Db4objects.Drs.Tests.Foundation.ObjectSetCollection4FacadeTestCase)
				).Run();
		}

		public virtual void TestEmpty()
		{
			Db4objects.Drs.Foundation.ObjectSetCollection4Facade facade = new Db4objects.Drs.Foundation.ObjectSetCollection4Facade
				(new Db4objects.Db4o.Foundation.Collection4());
			Db4oUnit.Assert.IsFalse(facade.HasNext());
			Db4oUnit.Assert.IsFalse(facade.HasNext());
		}

		public virtual void TestIteration()
		{
			Db4objects.Db4o.Foundation.Collection4 collection = new Db4objects.Db4o.Foundation.Collection4
				();
			collection.Add("bar");
			collection.Add("foo");
			Db4objects.Drs.Foundation.ObjectSetCollection4Facade facade = new Db4objects.Drs.Foundation.ObjectSetCollection4Facade
				(collection);
			Db4oUnit.Assert.IsTrue(facade.HasNext());
			Db4oUnit.Assert.AreEqual("bar", facade.Next());
			Db4oUnit.Assert.IsTrue(facade.HasNext());
			Db4oUnit.Assert.AreEqual("foo", facade.Next());
			Db4oUnit.Assert.IsFalse(facade.HasNext());
			facade.Reset();
			Db4oUnit.Assert.AreEqual("bar", facade.Next());
			Db4oUnit.Assert.AreEqual("foo", facade.Next());
			Db4oUnit.Assert.IsFalse(facade.HasNext());
		}
	}
}

File: drs/src/test/com/db4o/drs/test/foundation/ObjectSetCollection4FacadeTestCase.java
/* Copyright (C) 2004 - 2008  Versant Inc.  http://www.db4o.com

This file is part of the db4o open source object database.

db4o is free software; you can redistribute it and/or modify it under
the terms of version 2 of the GNU General Public License as published
by the Free Software Foundation and as clarified by db4objects' GPL 
interpretation policy, available at
http://www.db4o.com/about/company/legalpolicies/gplinterpretation/
Alternatively you can write to db4objects, Inc., 1900 S Norfolk Street,
Suite 350, San Mateo, CA 94403, USA.

db4o is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License along
with this program; if not, write to the Free Software Foundation, Inc.,
59 Temple Place - Suite 330, Boston, MA  02111-1307, USA. */
package com.db4o.drs.test.foundation;

import com.db4o.drs.foundation.ObjectSetCollection4Facade;
import com.db4o.foundation.Collection4;

import db4ounit.*;

/**
 * @sharpen.ignore
 */
public class ObjectSetCollection4FacadeTestCase implements TestCase {
	
	public static void main(String[] args) {
		new ConsoleTestRunner(ObjectSetCollection4FacadeTestCase.class).run();
	}
	
	public void testEmpty() {
		ObjectSetCollection4Facade facade = new ObjectSetCollection4Facade(new Collection4());
		Assert.isFalse(facade.hasNext());
		Assert.isFalse(facade.hasNext());
	}
	
	public void testIteration() {
		Collection4 collection = new Collection4();		
		collection.add("bar");
		collection.add("foo");
		
		ObjectSetCollection4Facade facade = new ObjectSetCollection4Facade(collection);
		Assert.isTrue(facade.hasNext());
		Assert.areEqual("bar", facade.next());
		Assert.isTrue(facade.hasNext());
		Assert.areEqual("foo", facade.next());
		Assert.isFalse(facade.hasNext());
		
		facade.reset();
		
		Assert.areEqual("bar", facade.next());
		Assert.areEqual("foo", facade.next());
		Assert.isFalse(facade.hasNext());
	}

}
